climakitae/explore/agnostic.py

```diff
@@ -268,207 +268,256 @@ def create_conversion_function(lookup_tables):
-        0
-    ]  # Need all the values in order to create histogram + return values
+    Parameters
+    ----------
+    lat: float
+        Latitude for specific location of interest.
+    lon: float
+        Longitude for specific location of interest.
+    metric: str
+        The metric to aggregate the LOCA simulations by.
+    years: tuple
+        The lower and upper year bounds (inclusive) to extract LOCA simulation data by.
+    months: list, optional
+        Specific months of interest. The default is all months.
 
-    # Splitting up the models by various statistics
-    single_model_compute, multiple_model_compute = _split_compute(sorted_sims)
+    Returns
+    -------
+    single_stats: dict of str: xr.DataArray
+        Dictionary mapping string names of statistics to single simulation xr.DataArray objects.
+    multiple_stats: dict of str: xr.DataArray
+        Dictionary mapping string names of statistics to multiple simulations xr.DataArray objects.
+    results: xr.DataArray
+        Aggregated results of running the given metric on the lat/lon gridcell of interest. Results are also sorted in ascending order.
+    """
+    # Create selections object
+    selections = _create_lat_lon_select(lat, lon, metric, years)
+    # Runs calculations and derives statistics from LOCA data pulled via selections object
+    return _compute_selections_and_stats(selections, metric, years, months)
 
-    # Creating a dictionary of single stats names to the initial models from the dataset
-    single_model_stats = dict(
-        zip(
-            list(single_model_compute.keys()),
-            subset_data.sel(simulation=list(single_model_compute.values())),
-        )
-    )
 
-    # Creating a dictionary of stats that return multiple models to the initial models from the dataset
-    multiple_model_stats = {
-        k: subset_data.sel(simulation=v) for k, v in multiple_model_compute.items()
-    }
+def get_area_subset_loca(
+    area_subset, cached_area, metric, years, months=list(np.arange(1, 13))
+):
+    """
+    This function combines all available LOCA simulation data that is filtered on the `area_subset` (a string
+    from existing keys in Boundaries.boundary_dict()) and on one of the areas of the values in that
+    `area_subset` (`cached_area`). It then extracts this data across all SSP pathways for specific years/months,
+    and runs the passed in metric on all of this data. The results are then returned in 3 values, the first
+    as a dict of statistic names to xr.DataArray single simulation objects (i.e. median),
+    the second as a dict of statistic names to xr.DataArray objects consisting of multiple simulation objects (i.e. middle 10%),
+    and the last as a xr.DataArray of simulations' aggregated values sorted in ascending order.
 
-    # Returning models from stats and aggregated results
-    return (single_model_stats, multiple_model_stats, sorted_sims)
+    Parameters
+    ----------
+    area_subset: str

```

Comment:
Very open to thoughts about any components of this docstring, particularly around the descriptions of the `area_subset` and `cached_area`.

Reply:
I think you explained it well.